### PR TITLE
Add NCI Imaging Data Commons dataset

### PR DESCRIPTION
**Pull request by fedorov**
This is initial draft introducing [NCI Imaging Data Commons](https://portal.imaging.datacommons.cancer.gov/) dataset, in preparation to the public release.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by fedorov** [comment]
depends on #1843

**Comment by fedorov** [comment]
@chueatwork I am marking this as ready, but note that the IDC v14 release is not live yet - will be soon. Let me know how you would like to coordinate the timing.

File: datasets/nci-imaging-data-commons.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  - digital pathology
+  - radiology
+  - microscopy
+  - fluorescence imaging
+  - image processing
+  - machine learning
+License: https://fairsharing.org/FAIRsharing.0b5a1d
+Resources:
+  - Description: Data files in a public bucket
+    ARN: arn:aws:s3:::TBD
```
[the remark this answers is not in the document]
Comment (chueatwork) [reply datasets/nci-imaging-data-commons.yaml:29]: Just add 3 separate resources, see how SRA does it: https://registry.opendata.aws/ncbi-sra/

[the remark this answers is not in the document]
Comment (G-White-ISB) [reply datasets/nci-imaging-data-commons.yaml:29]: The region is us-east-1

[the remark this answers is not in the document]
Comment (fedorov) [reply datasets/nci-imaging-data-commons.yaml:29]: Thank you both - I will update the PR.

**Comment by fedorov** [comment]
@chueatwork it is finally green!

**Comment by fedorov** [comment]
commits squashed to clean this up

**Pull request merged**